Gordon360/Services/DirectMessageService.cs

```diff
@@ -152,18 +152,33 @@ public List<Object> GetRoomById(string userId)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The data was not found." };
             }
-     
+
 
 
             return endresult;
 
         }
 
-        public bool CreateGroup(String id, String name, bool group, DateTime lastUpdated, string image)
+
+        public CreateGroupViewModel CreateGroup(String name, bool group, DateTime lastUpdated, string image, List<String> usernames)
         {
             DateTime createdAt = DateTime.Now;
+            Guid id = Guid.NewGuid();
 
-            var idParam = new SqlParameter("@_id", id);
+            var idParam = new SqlParameter("@_id", id.ToString());
+
+            var idCheck = RawSqlQuery<IdCheckViewModel>.query("CHECK_ID @_id", idParam);
+
+            foreach( IdCheckViewModel isNotAvailable in idCheck)
+            {
+                while (isNotAvailable.IdCheck)
+                {
+                    id = Guid.NewGuid();
+                    idParam = new SqlParameter("@_id", id.ToString());
+                    idCheck = RawSqlQuery<IdCheckViewModel>.query("CHECK_ID @_id", idParam);
+                }
+            }
+            var idRefreshParam = new SqlParameter("@_id", idParam.Value);
```

Comment:
That's good to know just took a look at it and it would be way more intuitive if we did that. I'm gonna see how much work it would be to change it to use this. Right now most of our functions expect our IDs to be strings that include numbers and letters.